papaya/src/main/java/com/palantir/util/TextUtils.java

```diff
@@ -0,0 +1,2041 @@
+                // looks like AA102
+                if (Character.isLetter(s.charAt(0)) && Character.isLetter(s.charAt(1))
+                        && (Character.isDigit(s.charAt(2)) || Character.isLetter(s.charAt(2)))
+                        && Character.isDigit(s.charAt(3)) && Character.isDigit(s.charAt(4))) {
+                    return s;
+                }
+            }
+            sNew.append(s.substring(0, 1).toUpperCase());
+            if (s.length() > 1) {
+                sNew.append(s.substring(1).toLowerCase());
+            }
+        }
+        return sNew.toString();
+    }
+
+
+    /**
+     * Returns either &quot;a&quot; or &quot;an&quot; to match the object.
+     */
+    // TODO (bcolwell) proper acronym support
+    public static String aan(final String object) {
+        if ( null == object || object.length() <= 0 ) {
+            return "a";
+        }
+
+        switch ( object.charAt( 0 ) ) {
+            case 'a':
+            case 'e':
+            case 'i':
+            case 'o':
+            case 'u':
+            case 'A':
+            case 'E':
+            case 'I':
+            case 'O':
+            case 'U':
+                return "an";
+            case 'S':
+                if ( object.toUpperCase().equals( object ) ) {
+                    // This case is an acronyn that starts with S:
+                    return "an";
+                }
+                break;
+        }
+        return "a";
+    }
+
+
+    /**
+     *  Makes a list of the n strings, separated by the given delimiter.
+     */
+
+    public static CharSequence makeStringList(List<?> strList, String delimiter) {
+        int count = strList.size();
+        if (count == 0) {
+            return "";
+        }
+
+        StringBuilder buf = new StringBuilder(10*count + delimiter.length()*(count-1));
+
+        for (int i = 0; i < count-1; i++) {
+            buf.append(strList.get(i));
+            buf.append(delimiter);
+        }
+        buf.append(strList.get(count-1));
+
+        return buf;
+    }
+
+
+    /**
+     *  Makes a list of n copies of the given string, separated by the given delimiter.
+     */
+
+    public static String makeStringList(String str, String delimiter, int count) {
+        if (count == 0) {
+            return "";
+        }
+        StringBuilder buf = new StringBuilder(str.length()*count + delimiter.length()*(count-1));
+
+        for (int i = 0; i < count-1; i++) {
+            buf.append(str);
+            buf.append(delimiter);
+        }
+        buf.append(str);
+
+        return buf.toString();
+    }
+
+    public static final String SPECIFIER_0 = "\\Q{0}\\E";
+    public static final String SPECIFIER_1 = "\\Q{1}\\E";
+    public static final String SPECIFIER_2 = "\\Q{2}\\E";
+    public static final String SPECIFIER_3 = "\\Q{3}\\E";
+    public static final String SPECIFIER_4 = "\\Q{4}\\E";
+
+    public static String format(String formatStr, String arg0) {
```

Comment:
It's pretty trivial to remove the entire papaya dependency here, we should do it.